vulnerabilities/importers/github.py

```diff
@@ -0,0 +1,342 @@
+
+            xml_resp = ET.ElementTree(ET.fromstring(resp.decode('utf-8')))
+            self.cache[pkg_name] = self.extract_versions(xml_resp)
+
+        except ET.ParseError:
+            self.cache[pkg_name] = set()
+
+    @staticmethod
+    def artifact_url(artifact_comps: List[str]) -> str:
+
+        base_url = 'https://repo.maven.apache.org/maven2/{}'
+        group_id, artifact_id = artifact_comps
+        group_url = group_id.replace('.', '/')
+        suffix = group_url + '/' + artifact_id + '/' + 'maven-metadata.xml'
+        endpoint = base_url.format(suffix)
+
+        return endpoint
+
+    @staticmethod
+    def extract_versions(xml_response: ET.ElementTree) -> Set[str]:
+
+        all_versions = set()
+        for child in xml_response.getroot().iter():
+            if child.tag == 'version':
+                all_versions.add(child.text)
+
+        return all_versions
+
+
+class NugetVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name.lower(), set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+        if pkg_name in self.cache:
+            return
+        endpoint = self.nuget_url(pkg_name)
+        try:
+            resp = requests.get(endpoint).json()
+        # pkg_name=Microsoft.NETCore.UniversalWindowsPlatform triggers
+        # JSONDecodeError.
+        except json.decoder.JSONDecodeError:
+            self.cache[pkg_name.lower()] = set()
+            return
+
+        self.cache[pkg_name.lower()] = self.extract_versions(resp)
+
+    @staticmethod
+    def nuget_url(pkg_name: str) -> str:
+        base_url = 'https://api.nuget.org/v3/registration5-semver1/{}/index.json'
+        return base_url.format(pkg_name.lower())
+
+    @staticmethod
+    def extract_versions(json_resp: dict) -> Set[str]:
+        all_versions = set()
+        try:
+            for entry in json_resp['items'][0]['items']:
+                all_versions.add(entry['catalogEntry']['version'])
+        # json response for YamlDotNet.Signed triggers this exception
+        except KeyError:
+            return all_versions
+
+        return all_versions
+
+
+class ComposerVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name.lower(), set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+        if pkg_name in self.cache:
+            return
+        endpoint = self.composer_url(pkg_name)
+        json_resp = requests.get(endpoint).json()
+        self.cache[pkg_name] = self.extract_versions(json_resp, pkg_name)
+
+    @staticmethod
+    def composer_url(pkg_name: str) -> str:
+        vendor, name = pkg_name.split('/')
+        return f'https://repo.packagist.org/p/{vendor}/{name}.json'
+
+    @staticmethod
+    def extract_versions(json_resp: dict, pkg_name: str) -> Set[str]:
+        all_versions = json_resp['packages'][pkg_name].keys()
```

Comment:
What has this response data mapping anything to do with JSON at this stage of processing?

Reply:
Renaming this `json_resp` to `resp`